src/CodeStyle/CSharp/Tests/FormattingAnalyzerTests.cs

```diff
@@ -0,0 +1,198 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Testing;
+using Microsoft.CodeAnalysis.Testing.Verifiers;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CodeStyle
+{
+    using Verify = CSharpCodeFixVerifier<CSharpFormattingAnalyzer, FormattingCodeFixProvider, XUnitVerifier>;
+
+    public class FormattingAnalyzerTests
+    {
+        [Fact]
+        public async Task TestAlreadyFormatted()
+        {
+            var testCode = @"
+class MyClass
+{
+    void MyMethod()
+    {
+    }
+}
+";
+
+            await Verify.VerifyAnalyzerAsync(testCode);
+        }
+
+        [Fact]
+        public async Task TestNeedsIndentation()
+        {
+            var testCode = @"
+class MyClass
+{
+  $$void MyMethod()
+  $${
+  $$}
```

Comment:
does this represent a vertical selection?

Reply:
No. There is no selection involved in analyzer tests; this is three distinct diagnostics reported in the input code.